src/subspace_store.c

```diff
@@ -104,50 +105,79 @@ subspace_store_add(SubspaceStore *store, const Hypercube *hc,
 
 		if (node == NULL)
 		{
+			/*
+			 * We should have one internal node per dimension in the
+			 * hypertable. If we don't have one for the current dimension,
+			 * create one now. (There will always be one for time)
```

Comment:
I think the only place we care about open vs closed is when evicting: our strategy of always picking the first element from the first dimension, isn't ideal for closed dimension.

Reply:
So, that is the point. It might not be ideal for the use case, because you'd like to evict along a time-dimension, removing old slices, rather than a particular space partition in a "current" time slice. But that's just the best way to structure dimensions in the subspace store for a particular use case and shouldn't affect the subspace store implementation. I can imagine us supporting only using hash partitioning (without time partitioning) in the future, because the code is mostly general enough to support that (modulo some hard-coded assumptions in places that should be easy to remove). That's why I am pushing back a bit on the formulation here. 

I think we can say in `ChunkDispatch`, where we actually use the SubspaceStore for a particular implementation of a state cache, that we make sure the open dimension should be first to efficiently evict old entries along the time dimension.

Reply:
I think I agree (though if we're going so general as to not have time dimensions we might want to implement regular LRU), but I don't think that's relevant to this PR, which should just be a bugfix, and not change the `SubspaceStore`'s semantics. Since the `SubspaceStore` guaranteed that the first dimension was a time-dimension going in, the PR should maintain that invariant, and we can generalize `SubspaceStore` in a future PR.

Reply:
Agreed, just don't want the docs to tie us further to particular restrictions that we don't need.